src/dependency-manager/spec/component-spec.ts

```diff
@@ -234,19 +247,20 @@ export class ComponentSpec {
     type: 'object',
 
     patternProperties: {
-      [ComponentSlugUtils.Validator.source]: {
+      [ComponentSlugUtils.Validator.source]: AnyOf({
         type: 'string',
         pattern: Slugs.ComponentTagValidator.source,
-      },
+      }, DependencySpec),
     },
 
     errorMessage: {
       additionalProperties: ComponentSlugUtils.Description,
     },
 
-    description: 'A key-value set of dependencies and their respective tags. Reference each dependency by component name (e.g. `cloud: latest`)',
+    description: 'A key-value set of dependencies with an empty value. Reference each dependency by component name (e.g. `cloud: {}`)',
   })
-  dependencies?: Dictionary<string>;
+  @Transform(transformDependencySpec(DependencySpec))
```

Comment:
This isn't quite what I meant. The ComponentSpec is supposed to be 1:1 with the architect.yml spec. We have the ComponentConfig to convert the ComponentSpec into something that is easier work with. So I was envisioning the transform would happen when we go from Spec -> Config. 

The ComponentSpec/ComponentConfig concept has its problems, but I don't know if we want to start transforming in multiple places without a larger overhaul.

Reply:
Ahh I see. So one thing I noticed was `ComponentConfig` is very rarely used - in the API and this PR where I had to check the whole "tag_or_object" thing, it was a `ComponentSpec` object that was being used so the `ComponentConfig` doing a transformation wouldn't have actually helped at all (so I assumed you meant the spec 😛 )

Reply:
I can undo this change and instead have the transform happen just for `ComponentConfig`, it just means that we still need to check if `tag_or_object` is a string or an object in the `register` command for example

Reply:
node.config is the ComponentConfig which should be used more, but we def haven't realized the initial vision...